File: sample-java/app/src/main/java/com/seerslab/argear/sample/viewmodel/ContentsViewModel.java
package com.seerslab.argear.sample.viewmodel;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.seerslab.argear.sample.AppConfig;
import com.seerslab.argear.sample.api.ContentsResponse;

public class ContentsViewModel extends AndroidViewModel {

    private MutableLiveData<ContentsResponse> mutableLiveData;
    private ContentsRepository contentsRepository;

    public ContentsViewModel(Application application) {
        super(application);
        contentsRepository = ContentsRepository.getInstance();
        mutableLiveData = contentsRepository.getContents(AppConfig.API_KEY);
    }

    public LiveData<ContentsResponse> getContents() {
        return mutableLiveData;
    }
}

File: sample-kotlin/app/src/main/java/com/seerslab/argear/sample/viewmodel/ContentsViewModel.kt
package com.seerslab.argear.sample.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.seerslab.argear.sample.AppConfig
import com.seerslab.argear.sample.api.ContentsResponse

class ContentsViewModel(application: Application) : AndroidViewModel(application) {

    private val mutableLiveData: MutableLiveData<ContentsResponse>
    private val contentsRepository: ContentsRepository = ContentsRepository.instance

    val contents: LiveData<ContentsResponse>
        get() = mutableLiveData

    init {
        mutableLiveData = contentsRepository.getContents(AppConfig.API_KEY)
    }
}